docs/how-to/unified_memory.rst

```diff
@@ -0,0 +1,638 @@
+hint techniques are presented in this section.
+
+The hint functions can set actions on a selected device, which can be
+identified by ``hipGetDeviceProperties(&prop, device_id)``. There are two
+special ``device_id`` values:
+
+- ``hipCpuDeviceId`` = -1 means that the advised device is the CPU.
+- ``hipInvalidDeviceId`` = -2 means that the device is invalid.
+
+For the best performance you can profile your application to optimize the
+utilization of compiler hits.
+
+Data Prefetching
+----------------
+Data prefetching is a technique used to improve the performance of your
+application by moving data closer to the processing unit before it is actually
+needed.
+
+.. code-block:: cpp
+    :emphasize-lines: 20-23,31-32
+
+    // Addition of two values.
+    __global__ void add(int *a, int *b, int *c) {
+        *c = *a + *b;
+    }
+
+    int main() {
+        int *a, *b, *c;
+        int deviceId;
+        hipGetDevice(&deviceId); // Get the current device ID
+
+        // Allocate memory for a, b and c that is accessible to both device and host codes.
+        hipMallocManaged(&a, sizeof(*a));
+        hipMallocManaged(&b, sizeof(*b));
+        hipMallocManaged(&c, sizeof(*c));
+
+        // Setup input values.
+        *a = 1;
+        *b = 2;
+
+        // Prefetch the data to the GPU device.
+        hipMemPrefetchAsync(a, sizeof(*a), deviceId, 0);
+        hipMemPrefetchAsync(b, sizeof(*b), deviceId, 0);
+        hipMemPrefetchAsync(c, sizeof(*c), deviceId, 0);
+
+        // Launch add() kernel on GPU.
+        hipLaunchKernelGGL(add, dim3(1), dim3(1), 0, 0, a, b, c);
+
+        // Wait for GPU to finish before accessing on host.
+        hipDeviceSynchronize();
+
+        // Prefetch the result back to the CPU.
+        hipMemPrefetchAsync(c, sizeof(*c), hipCpuDeviceId, 0);
+
+        // Wait for the prefetch operations to complete.
+        hipDeviceSynchronize();
+
+        // Prints the result.
+        std::cout << *a << " + " << *b << " = " << *c << std::endl;
+
+        // Cleanup allocated memory.
+        hipFree(a);
+        hipFree(b);
+        hipFree(c);
+
+        return 0;
+    }
+
+Remember to check the return status of ``hipMemPrefetchAsync()`` to ensure that
+the prefetch operations complete successfully!
+
+Memory Advise
+-------------
+The effectiveness of ``hipMemAdvise()`` comes from its ability to inform the
+runtime system of the developer's intentions regarding memory usage. When the
+runtime system has knowledge of the expected memory access patterns, it can
+make better decisions about data placement and caching, leading to more
+efficient execution of the application. However, the actual impact on
+performance can vary based on the specific use case and the hardware
+architecture.
+
+For the description of ``hipMemAdvise()`` and the detailed list of advises,
+visit the :ref:`unified_memory_reference`.
+
+Here is the updated version of the example above with memory advises.
```

Comment:
is "advise" the right term?
perhaps "advice"
seems odds

Reply:
Thanks! I fixed all occurrences of the noun `advice`.